pkg/planner/core/indexmerge_path.go

```diff
@@ -148,143 +152,17 @@ func generateNormalIndexPartialPath(
 			needSelection = true
 		}
 	}
-	itemPaths := accessPathsForConds(ds, pushedDownCNFItems, []*util.AccessPath{candidatePath})
-	if len(itemPaths) != 1 {
-		// for this dnf item, we couldn't generate an index merge partial path.
-		// (1 member of (a)) or (3 member of (b)) or d=1; if one dnf item like d=1 here could walk index path,
-		// the entire index merge is not valid anymore.
-		return nil, false
-	}
-	partialPath := buildIndexMergePartialPath(itemPaths)
+	partialPath := accessPathsForConds(ds, pushedDownCNFItems, candidatePath)
 	if partialPath == nil {
 		// for this dnf item, we couldn't generate an index merge partial path.
 		// (1 member of (a)) or (3 member of (b)) or d=1; if one dnf item like d=1 here could walk index path,
 		// the entire index merge is not valid anymore.
 		return nil, false
 	}
 
-	// identify whether all pushedDownCNFItems are fully used.
-	// If any partial path contains table filters, we need to keep the whole DNF filter in the Selection.
-	if len(partialPath.TableFilters) > 0 {
-		needSelection = true
-		partialPath.TableFilters = nil
-	}
-	// If any partial path's index filter cannot be pushed to TiKV, we should keep the whole DNF filter.
-	if len(partialPath.IndexFilters) != 0 && !expression.CanExprsPushDown(pushDownCtx, partialPath.IndexFilters, kv.TiKV) {
-		needSelection = true
-		// Clear IndexFilter, the whole filter will be put in indexMergePath.TableFilters.
-		partialPath.IndexFilters = nil
-	}
-	// Keep this filter as a part of table filters for safety if it has any parameter.
-	if expression.MaybeOverOptimized4PlanCache(ds.SCtx().GetExprCtx(), cnfItems) {
-		needSelection = true
-	}
-
 	return partialPath, needSelection
 }
 
-// getIndexMergeOrPath generates all possible IndexMergeOrPaths.
-// For index merge union case, the order property from its partial
-// path can be kept and multi-way merged and output. So we don't
-// generate a concrete index merge path out, but an un-determined
-// alternatives set index merge path instead.
-//
-//	    `create table t (a int, b int, c int, key a(a), key b(b), key ac(a, c), key bc(b, c))`
-//		`explain format='verbose' select * from t where a=1 or b=1 order by c`
-//
-// like the case here:
-// normal index merge OR path should be:
-// for a=1, it has two partial alternative paths: [a, ac]
-// for b=1, it has two partial alternative paths: [b, bc]
-// and the index merge path:
-//
-//	indexMergePath: {
-//	    PartialIndexPaths: empty                          // 1D array here, currently is not decided yet.
-//	    PartialAlternativeIndexPaths: [[a, ac], [b, bc]]  // 2D array here, each for one DNF item choices.
-//	}
-func generateIndexMergeOrPaths(ds *logicalop.DataSource, filters []expression.Expression) error {
```

Comment:
Moved to `pkg/planner/core/indexmerge_unfinished_path.go` as `generateORIndexMerge()`.